lib/mix/tasks/guardian_db.gen.migration.ex

```diff
@@ -0,0 +1,43 @@
+defmodule Mix.Tasks.GuardianDb.Gen.Migration do
+  @shortdoc "Generates GuardianDb's migration"
+
+  @moduledoc """
+  Generates the required GuardianDb's database migration
+  """
+  use Mix.Task
+
+  alias Mix.Generator
+
+  @doc false
+  def run(_args) do
+    if Mix.Project.umbrella?() do
+      Mix.raise "mix guardian_db.gen.migration can only be run inside an application directory"
+    end
+
+    Generator.create_directory("./priv/repo/migrations")
```

Comment:
What happen if already exists?

Also `/repo` is base on the Ecto repo so if my repo is not `Repo` then this will be wrong.

For example

```elixir
defmodule MyApp.OtherNameRepo do
  # ...
end
```

The migration folder for this will be `./priv/other_name_repo/...`

I think you need some `-r` like other Ecto tasks do where you request the to pass the repo, if not then fallbacks to the default one.

I am not sure how Ecto handles it but take a look to some ecto task.

Reply:
Interesting, I'll take a look at that option.

Reply:
@rhnonose take a look to migration task https://github.com/elixir-ecto/ecto/blob/master/lib/mix/tasks/ecto.gen.migration.ex#L41 

@rhnonose this is the folder path https://github.com/elixir-ecto/ecto/blob/master/lib/mix/tasks/ecto.gen.migration.ex#L49

Probably just import a bunch of utils from  `import Mix.Ecto`